### ISSUE TITLE
Set reporter programmatically

### ISSUE THREAD

**Issue by 0x1mason** [avajs/gulp-ava#17]
Possible via options?

**Comment by 0x1mason** [comment]
Custom reporter, e.g. tap-nyan.

**Comment by sindresorhus** [comment]
From the readme:

> Adheres to AVA options in package.json. **You can also specify options in the plugin as seen above.**

**Comment by 0x1mason** [comment]
@sindresorhus Thanks for the quick response. I read the docs and see the options. I do not see an option for reporters. Does that mean there isn't one?

**Comment by 0x1mason** [comment]
The docs only mention piping to custom reporters via the the cli which is why I'm inquiring here. Is there a gulp specific workaround?

**Comment by sindresorhus** [comment]
Yeah, you're right. Not that clear. AVA doesn't support third-party reporters directly, but instead comes with a TAP option, enabling you to use any [TAP reporter](https://github.com/avajs/ava#tap-reporter). See the `"tap": true,` option. Although, thinking about it more, the Gulp output might conflict with the TAP output, so I'm not 100% sure it will work correctly. I think TAP is supposed to ignore non-TAP output, but needs to be confirmed.

That being said. Would honestly be easier to just spawn the `ava` binary directly from Gulp.

**Comment by 0x1mason** [comment]
Cool, thanks. Yeah, I'll try spawning or hacking the plugin.

**Comment by sindresorhus** [comment]
Not tested, but with [`execa`](https://github.com/sindresorhus/execa) it could probably be as simple as:

```js
gulp.task('test', () => execa.shell('ava --tap | tap-nyan', {stdio: 'inherit'}));
```
